### PR TITLE
Preload large Bramble assets on homepage in order to cache

### PR DESCRIPTION
**Pull request by humphd**
This fixes how I was doing the pre-fetching of Bramble's largest assets.  This is going to need a CORS update to our staging and prod S3 buckets, in order to allow our various Thimble servers to xhr GET some files from the Bramble S3 bucket.  @jbuck, can you help with this, I followed http://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html#how-do-i-enable-cors:

```xml
<CORSConfiguration>
 <CORSRule>
   <AllowedOrigin>https://bramble.mofostaging.net</AllowedOrigin>
   <AllowedMethod>GET</AllowedMethod>
   <AllowedHeader>*</AllowedHeader>
 </CORSRule>
 <CORSRule>
   <AllowedOrigin>https://thimble-beta.webmaker.org</AllowedOrigin>
   <AllowedMethod>GET</AllowedMethod>
   <AllowedHeader>*</AllowedHeader>
 </CORSRule>
 <CORSRule>
   <AllowedOrigin>https://thimble.mozilla.org</AllowedOrigin>
   <AllowedMethod>GET</AllowedMethod>
   <AllowedHeader>*</AllowedHeader>
 </CORSRule>
</CORSConfiguration>
```

### CODE REVIEW

File: public/homepage/scripts/main.js [hunk 1]
```diff
@@ -21,6 +21,20 @@ require.config({
   }
 });
 
+// While the user is reading this page, start to cache Bramble's biggest files
+function preloadBramble($) {
+  var brambleHost = $("meta[name='bramble-host']").attr("content");
+  [
+    brambleHost + "/dist/styles/brackets.min.css",
+    brambleHost + "/dist/bramble.js",
```
Comment (ashleygwilliams) [inline public/homepage/scripts/main.js]: can we concat and minify all this js?

**Comment by sedge** [comment]
r+, nice one. The XML looks correct too.

File: public/homepage/scripts/main.js (hunk 1 above)
Comment (humphd) [inline public/homepage/scripts/main.js]: It is already.

**Comment by humphd** [comment]
@jbuck, I get XHR CORS errors with https://bramble.mofostaging.net/?aa;ldskfjs;dlfkjasd;lfkj:

```
XMLHttpRequest cannot load https://mozillathimblelivepreview.net/bramble/staging/dist/styles/brackets.min.css. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://bramble.mofostaging.net' is therefore not allowed access.
bramble.mofostaging.net/:1 XMLHttpRequest cannot load https://mozillathimblelivepreview.net/bramble/staging/dist/bramble.js. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://bramble.mofostaging.net' is therefore not allowed access.
bramble.mofostaging.net/:1 XMLHttpRequest cannot load https://mozillathimblelivepreview.net/bramble/staging/dist/thirdparty/thirdparty.min.js. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://bramble.mofostaging.net' is therefore not allowed access.
bramble.mofostaging.net/:1 XMLHttpRequest cannot load https://mozillathimblelivepreview.net/bramble/staging/dist/main.js. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://bramble.mofostaging.net' is therefore not allowed access.
```

**Comment by humphd** [comment]
I'm going to revert this, we're not going to get this to work with our current infra setup.

**Pull request merged**